### PR TITLE
fix: transactions details UI

### PR DESCRIPTION
**Pull request by ice-brontes**
## Description
I added an in-app browser to display the explore link as a modal dialog. Also, added small fixes on the transaction details page.

## Type of Change
- [x] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Refactoring
- [ ] Documentation
- [ ] Chore

## Video presentation
https://github.com/user-attachments/assets/d72c9193-d0d0-4cca-91b8-e32433d9231f

### CODE REVIEW

File: lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  const ExploreTransactionDetailsModal({
+    required this.url,
+    super.key,
+  });
+
+  final String url;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final webViewController = useState<InAppWebViewController?>(null);
```
Comment (ice-tychon) [inline lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart:25]: Should it be `useState` or `useRef`? Do we need to rebuild the widget when the value is set?

File: lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    return SheetContent(
+      body: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          NavigationAppBar.modal(
+            title: Text(context.i18n.wallet_explore_transaction_details_title),
+            actions: const [NavigationCloseButton()],
+          ),
+          Expanded(
+            child: Stack(
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart:38]: To display the progress indicator in front of the web view. I will move it in one custom web view widget.

File: lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                    webViewController.value = controller;
+                  },
+                  onLoadStart: (controller, url) {
+                    isLoadingPage.value = false;
+                  },
+                  onLoadStop: (controller, url) {
+                    isLoadingPage.value = true;
+                  },
+                ),
+                ValueListenableBuilder(
```
Comment (ice-tychon) [inline lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart:52]: If we use `useState` for `isLoadingPage` the whole widget is gonna be rebuilt, no need to wrap the loaded with `ValueListenableBuilder`.

File: lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          NavigationAppBar.modal(
+            title: Text(context.i18n.wallet_explore_transaction_details_title),
+            actions: const [NavigationCloseButton()],
+          ),
+          Expanded(
+            child: Stack(
+              children: [
+                InAppWebView(
```
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart]: Moved to `WebView` widget.

File: lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart]: Nope, removed. Left after testing phase.

Comment (ice-brontes) [reply lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart]: Removed this controller.

File: lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (ice-brontes) [reply lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart]: Removed `ValueListenableBuilder`.

Comment (ice-brontes) [reply lib/app/features/wallet/views/pages/transaction_details/explore_transaction_details_modal.dart]: Removed `ValueListenableBuilder` here.

**Review by ice-alcides — APPROVED** [review]

**Review by ice-tychon — APPROVED** [review]

**Pull request merged**